service/controller/resource/subnet/error.go

```diff
@@ -0,0 +1,59 @@
+package subnet
+
+import (
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/giantswarm/microerror"
+)
+
+var invalidConfigError = &microerror.Error{
+	Kind: "invalidConfigError",
+}
+
+var notFoundError = &microerror.Error{
+	Kind: "notFoundError",
+}
+
+// IsNotFound asserts notFoundError.
+func IsNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	c := microerror.Cause(err)
+
+	if c == notFoundError {
+		return true
+	}
+
+	{
+		dErr, ok := c.(autorest.DetailedError)
+		if ok {
+			if dErr.StatusCode == 404 {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+var missingOutputValueError = &microerror.Error{
+	Kind: "missingOutputValueError",
+}
+
+// IsMissingOutputValue asserts missingOutputValueError.
+func IsMissingOutputValue(err error) bool {
+	return microerror.Cause(err) == missingOutputValueError
+}
+
+var tooManyCredentialsError = &microerror.Error{
```

Comment:
I've been told that we always need those error matcher functions (like `IsMissingOutputValue` here) for every type of error we add (even though it is not used :shrug:).

Reply:
![](https://media1.giphy.com/media/zNbiX43QsqUAU/200.gif)